scvi/dataset/synthetic.py

```diff
@@ -42,3 +42,113 @@ def __init__(self, mu=4.0, theta=2.0, dropout=0.7, save_path='data/'):
         super().__init__(
             *GeneExpressionDataset.get_attributes_from_matrix(X_train, labels=labels_simlr)
         )
+
+
+class SyntheticDatasetCorrelated(GeneExpressionDataset):
+    def __init__(self, n_cells_cluster=100, n_clusters=5,
+                 n_genes_high=15, n_genes_total=50,
+                 weight_high=0.1, weight_low=0.01,
+                 lam_0=100., p_dropout=0.2,
+                 n_batches=1, n_labels=1, mode='mixed', ratio_genes_zi=0.5):
+        """
+
+        :param n_cells_cluster: Number of cells in each cluster
+        :param n_clusters: Number of cell clusters
+        :param n_genes_high: Number of highly expressed genes for each cluster.
+        Other genes are still expressed but at a lower expression
+        The level expression for highly expressed genes follow the same distribution
+
+        :param n_genes_total: Number of genes in total
+        :param weight_high: Level weight for highly expressed genes
+        :param weight_low: Level weight for lowly expressed genes
+        :param lam_0: Proportionality in the Poisson distributions parameter
+        :param p_dropout: Probability of dropout (only makes sense if mode='zi'
+        or mode='mixed'
+        :param n_batches: (useless for now) Number of batches in dataset
+        :param n_labels: (useless for now) Number of labels
+        :param mode: Should be either
+            - 'nb': mode without zero inflation
+            - 'zi': mode including zero inflation for ALL genes
+            -'mixed': mode in which SOME genes are Zero-Inflated while some other are not
```

Comment:
Do we need "mixed" mode?

Reply:
Well, it may be very specific to our work on comparing ZINB and NB models.
Basically, the mixed classes would allow us to verify that our approach is able to differentiate between ZI and non-ZI genes.
Maybe this mode is not relevant for scVI's general purpose and tasks though

Reply:
It's probably fine to include it if it's relevant for your experiments. I just didn't realize that mixed mode was a realistic scenario.

Rather than having separate "mixed" and "zi" modes, how about using the ZIFA dropout model to create this synthetic data, and letting the user specify the rate parameters lambda? I think that could give use more realistic dropout. Almost none of the highly expressed genes will be dropped out then, and many of the infrequently expressed genes will be dropped out for some settings of lambda.